packages/paste-core/components/button/src/index.tsx

```diff
@@ -42,23 +42,43 @@ const getButtonState = ({loading, disabled}: ButtonProps): ButtonStates => {
   return 'default';
 };
 
-const Button: React.FC<ButtonProps> = props => {
-  const buttonState = getButtonState(props);
+const Button: React.FC<ButtonProps> = ({
+  as,
+  autoFocus,
+  children,
+  disabled,
+  fullWidth,
+  href,
+  loading,
+  onBlur,
+  onClick,
+  onFocus,
+  onMouseDown,
+  onMouseEnter,
+  onMouseLeave,
+  onMouseUp,
+  size,
+  tabIndex,
+  type,
+  variant,
+  ...props
+}) => {
+  const buttonState = getButtonState({loading, disabled, variant, children});
```

Comment:
For these, lets not re-wrap them in an object. Let’s just pass comma separated arguments and modify the functions to not use `props` (or what ever it uses)

Reply:
Restructured. Let me know if I have that setup correctly.